### PR TITLE
[SPARK-34597][SQL] Replaces `ParquetFileReader.readFooter` with `ParquetFileReader.open and getFooter`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
`ParquetFileReader.readFooter` related methods has been identified as `@Deprecated` and `Apache Parquet` suggests replace it with the combination of `ParquetFileReader.open() and getFooter()` methods.

This PR introduces the `ParquetFooterReader` utility class due to some repetitive code patterns when read parquet file footer.

### Why are the changes needed?
Cleanup deprecated API usage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass the Jenkins or GitHub Action

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @sunchao and @maropu

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetFooterReader.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.apache.parquet.ParquetReadOptions;
+import org.apache.parquet.format.converter.ParquetMetadataConverter;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+
+import java.io.IOException;
+
+/**
+ * `ParquetFooterReader` is a Util class which encapsulates the helper
```
Comment (sunchao) [inline sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetFooterReader.java:33]: nit: `Util` -> `util`, `helper method` -> `help methods`.

**Review by sunchao — APPROVED** [review]
LGTM (non-binding) with one nit. Thanks @LuciferYang !

**Comment by maropu** [comment]
The code change itself looks fine. Checking the performance change now just in case.

**Comment by dongjoon-hyun** [comment]
> The code change itself looks fine. Checking the performance change now just in case.

Thank you so much always, @maropu .

**Comment by dongjoon-hyun** [comment]
Thank you, @sunchao , too!

**Review by maropu — APPROVED** [review]
okay, I've checked that there is no perf. regression of TPCDS queries. cc: @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Thank you, @maropu .

@LuciferYang . Could you address @sunchao 's comment, https://github.com/apache/spark/pull/31711#discussion_r588070563?

**Comment by LuciferYang** [comment]
Thank you for your review ~ @maropu @dongjoon-hyun  @sunchao,  279d7c3 fix @sunchao's comment ~

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @LuciferYang and all.
Merged to master!

**Comment by LuciferYang** [comment]
Thx ~ @dongjoon-hyun

**Pull request closed without merging**